frontend/src/api/RestaurantApi.tsx

```diff
@@ -0,0 +1,43 @@
+import { SearchState } from '@/pages/SearchPage';
+import { RestaurantResponse } from '@/types';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useQuery } from 'react-query';
+
+const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
+
+export const useSearchRestaurants = (
+  searchState: SearchState,
+  city: string
+) => {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const createSearchRestaurant = async (): Promise<RestaurantResponse> => {
+    const accessToken = await getAccessTokenSilently();
+    const params = new URLSearchParams();
+    params.set('searchQuery', searchState.searchQuery);
+
+    const response = await fetch(
+      `${API_BASE_URL}/api/restaurant/search/${encodeURIComponent(city)}?${params.toString()}`,
+      {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    );
+
+    if (!response.ok) {
+      throw new Error('Failed to search restaurants');
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/43#issuecomment-2408154714>#86982f</a><span> - 10/11/2024, 09:46 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The error handling in the createSearchRestaurant function is too generic. It doesn't provide enough information about the nature of the error, which could make debugging difficult.
</div>
<div id="fix">
<b>Fix:</b> Enhance the error handling to provide more specific error messages. Catch different types of errors (network errors, API errors, etc.) and provide appropriate error messages for each.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -27,7 +27,15 @@
      );
 
      if (!response.ok) {
 -      throw new Error('Failed to search restaurants');
 +      const errorBody = await response.text();
 +      if (response.status === 404) {
 +        throw new Error(`No restaurants found in ${city}`);
 +      } else if (response.status === 401) {
 +        throw new Error('Unauthorized access. Please log in again.');
 +      } else if (response.status >= 500) {
 +        throw new Error(`Server error: ${errorBody}`);
 +      } else {
 +        throw new Error(`Failed to search restaurants: ${errorBody}`);
 +      }
      }
 
      return response.json();
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged